homeassistant/components/google_tasks/application_credentials.py

```diff
@@ -0,0 +1,23 @@
+"""application_credentials platform the Google Tasks integration."""
+
+from homeassistant.components.application_credentials import AuthorizationServer
+from homeassistant.core import HomeAssistant
+
+from .const import OAUTH2_AUTHORIZE, OAUTH2_TOKEN
+
+
+async def async_get_authorization_server(hass: HomeAssistant) -> AuthorizationServer:
+    """Return authorization server."""
+    return AuthorizationServer(
+        authorize_url=OAUTH2_AUTHORIZE,
```

Comment:
Side note: Shared application credentials between integrations came to mind when I read this. I don't remember what we said about this in the architecture discussion.

Reply:
Following up, https://github.com/home-assistant/core/pull/103570 is an attempt to make google calendar support web auth so it could have compatible creds with other google integrations.